### DISCUSSION TITLE
How to set imported avatar to humanoid

### DISCUSSION THREAD

**Discussion by yuchenz27** [atteneder/glTFast#359] [Q&A] [answered]
I want to set the imported avatar model to humanoid.

If we import a FBX model, we can manually set the animation type to be humanoid like this
![image](https://user-images.githubusercontent.com/44870300/156914885-cdf112ff-040a-4614-b3aa-f254b9d1202e.png)
 
But when I imported a GLB file, I got this
<img width="408" alt="image" src="https://user-images.githubusercontent.com/44870300/156914941-dfb2565e-807c-40dc-bddb-881a3d2399c5.png">

How can I set the imported GLB model to be humanoid just like FBX models? Thank you.

**Comment by atteneder** [comment, marked as answer]
@yuchenz27 Thanks for this suggestion!

Currently glTFast does not offer most of the importer settings the formats with out-of-the-box support offer. Unfortunately these are not (easily) re-usable for ScriptedImporters, so we'd basically have to rewrite them. I do consider this, but probably starting with material extraction first and looking into special use-cases like humanoid rigs later.

A more glTF-like solution would be drafting or supporting glTF extensions that bring humanoid rig capabilities to glTF itself. You should have a look at [VRM](https://vrm.dev/), which is a glTF based format specifically for humanoid avatars. It comes with a Unity package as well. It's not using glTFast (it has its own glTF importer) and I haven't had a close look at it. I'd like to do so at some point and maybe add VRM support to glTFast as well.

**Reply by yuchenz27** [reply]
Thank you very much! I will have a look at VRM.

**Comment by smith076** [comment]
I had the same issue today, I dropped the model in the scene and exported that model as an fbx. Worked fine.
